safe-global/safe-wallet-web
# [Counterfactual] Show notification when submitting counterfactual tx

## What it solves

Resolves #

## How this PR fixes it

- Closes the deploy safe tx-flow when submitting the transaction
- Shows a notification when submitting the deploy safe transaction
- Shows an info screen on the first-tx + deploy flow breaking down the cost of the transaction
- Fixes an issue where the counterfactual state was cleared even if the transaction failed
- Estimates the actual gas costs of the signed transaction and uses that instead of what `useDeployGasLimit` returns

## How to test it

1. Create a counterfactual safe
2. Choose the Activate now option from settings
3. Submit the transaction
4. Observe the modal closes
5. Observe there is a notification
6. Deploy the safe with the first transaction
7. Observe there is an info screen visible
8. Cancel the transaction in your wallet
9. Observe an error notification shown
10. The counterfactual state should not be cleared in that case

## Screenshots

<img width="365" alt="Screenshot 2024-02-08 at 17 54 18" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/a772fa2d-5724-424a-a0cd-2f61b85d3c6b">
<img width="706" alt="Screenshot 2024-02-09 at 12 29 55" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/9d1b0a12-85e9-4293-80db-f37dd0888cd9">


## Checklist
